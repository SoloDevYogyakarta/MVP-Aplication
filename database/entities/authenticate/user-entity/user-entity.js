"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.userEntity = exports.UserEntity = void 0;
const sequelize_1 = require("sequelize");
const base_entity_1 = require("../../base-entity/base-entity");
const entity_1 = require("../../entity");
class UserEntity extends base_entity_1.BaseEntity {
}
exports.UserEntity = UserEntity;
exports.userEntity = entity_1.sequelize.define('USER', {
    plat_number: sequelize_1.DataTypes.STRING,
    phone_number: sequelize_1.DataTypes.STRING,
    name: sequelize_1.DataTypes.STRING,
    motor: sequelize_1.DataTypes.STRING,
    year_production: sequelize_1.DataTypes.INTEGER,
    address: sequelize_1.DataTypes.STRING,
    role: sequelize_1.DataTypes.STRING,
    password: sequelize_1.DataTypes.STRING,
}, {
    tableName: 'USER',
    schema: 'AUTHENTICATE',
});
//# sourceMappingURL=user-entity.js.map