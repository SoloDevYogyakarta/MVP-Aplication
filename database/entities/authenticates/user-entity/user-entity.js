"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.userEntity = exports.UserEntity = void 0;
const sequelize_1 = require("sequelize");
const nanoid_1 = require("nanoid");
const bcrypt_1 = __importDefault(require("bcrypt"));
const entity_1 = require("../../entity");
const base_entity_1 = require("../../base-entity/base-entity");
class UserEntity extends base_entity_1.BaseEntity {
}
exports.UserEntity = UserEntity;
exports.userEntity = entity_1.sequelize.define('USER', {
    public_id: sequelize_1.DataTypes.STRING,
    username: sequelize_1.DataTypes.STRING,
    email: sequelize_1.DataTypes.STRING,
    phone_number: sequelize_1.DataTypes.INTEGER,
    password: sequelize_1.DataTypes.STRING,
    file_id: sequelize_1.DataTypes.STRING,
}, {
    tableName: 'USER',
    schema: 'AUTHENTICATE',
    hooks: {
        beforeCreate(attributes, options) {
            const instance = attributes;
            instance.public_id = (0, nanoid_1.nanoid)();
            instance.password = bcrypt_1.default.hashSync(instance.password, 10);
        },
    },
});
//# sourceMappingURL=user-entity.js.map