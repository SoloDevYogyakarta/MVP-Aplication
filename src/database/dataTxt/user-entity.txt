{"id":244,"username":"Hattie_Waters23","password":"$2b$10$KkrxxQz79dBpcJHsOryBjeta/ltDU0p3vo2cTeYrBcBxpblrlmh86","updatedAt":"2023-10-21T09:54:52.365Z","createdAt":"2023-10-21T09:54:52.365Z","public_id":"ehRhduFOwW00p5qF18jzw","email":null,"phone_number":null}