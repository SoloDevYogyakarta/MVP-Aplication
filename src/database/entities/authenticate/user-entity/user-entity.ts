import { nanoid } from 'nanoid';
import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { SuperTest } from 'supertest';
import { BaseEntity } from '../../base-entity/base-entity';
import { sequelize } from '../../entity';

export class UserEntity extends BaseEntity {
  plat_number: string;
  phone_number: string;
  name: string;
  motor: string;
  year_production: number;
  address: string;
  role: string;
  password: string;
}

export type UserCreationAttribute = Optional<UserEntity, 'id'>;
export interface UserInstance
  extends Model<UserCreationAttribute, UserEntity>,
    UserEntity {}
export const userEntity = sequelize.define<UserInstance>(
  'USER',
  {
    plat_number: DataTypes.STRING,
    phone_number: DataTypes.STRING,
    name: DataTypes.STRING,
    motor: DataTypes.STRING,
    year_production: DataTypes.INTEGER,
    address: DataTypes.STRING,
    role: DataTypes.STRING,
    password: DataTypes.STRING,
  } as ModelAttributes<UserInstance>,
  {
    tableName: 'USER',
    schema: 'AUTHENTICATE',
  },
);
