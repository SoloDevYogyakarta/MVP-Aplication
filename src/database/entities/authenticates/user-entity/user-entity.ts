import { DataTypes, Model, ModelAttributes, Optional } from 'sequelize';
import { nanoid } from 'nanoid';
import bcrypt from 'bcrypt';
import { sequelize } from '../../entity';
import { BaseEntity } from '../../base-entity/base-entity';

export class UserEntity extends BaseEntity {
  username: string;
  email: string;
  phone_number: number;
  password: string;
  role: string;
  file_id: string;
}

export type UserCreationAttribute = Optional<UserEntity, 'id'>;
export interface UserInstance
  extends Model<UserCreationAttribute, UserEntity>,
    UserEntity {}
export const userEntity = sequelize.define<UserInstance>(
  'USER',
  {
    public_id: DataTypes.STRING,
    username: DataTypes.STRING,
    email: DataTypes.STRING,
    phone_number: DataTypes.INTEGER,
    role: DataTypes.STRING,
    password: DataTypes.STRING,
    file_id: DataTypes.STRING,
  } as ModelAttributes<UserInstance>,
  {
    tableName: 'USER',
    schema: 'AUTHENTICATE',
    hooks: {
      beforeCreate(attributes, options) {
        const instance = attributes;
        instance.public_id = nanoid();
        instance.password = bcrypt.hashSync(instance.password, 10);
      },
    },
  },
);
