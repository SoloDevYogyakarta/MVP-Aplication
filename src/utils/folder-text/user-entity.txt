{"id":1057,"plat_number":"B 6929 SHA","phone_number":"540-311-3034","name":"Yvette Tremblay","motor":"Sleek Soft Mouse","address":"39389 Franey Courts","year_production":2861,"role":"admin","password":"$2b$10$Q3pPGa3yL/6IFbR/nUsj8uq2hFGtAbDTUkMqX5A.FyonjCXibU58K","updatedAt":"2023-11-06T15:51:08.666Z","createdAt":"2023-11-06T15:51:08.666Z"}