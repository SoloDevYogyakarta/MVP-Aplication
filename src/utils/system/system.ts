import fs from 'fs';
import { join } from 'path';

export function joinpath(path: string): string {
  return join(__dirname, path);
}

export function createpath(path: string, data: unknown): void {
  if (typeof data !== 'string') {
    data = JSON.stringify(data);
  }
  fs.writeFileSync(joinpath(path), data as string);
}

export function readpath(path: string): string {
  return fs.readFileSync(joinpath(path), { encoding: 'utf-8' });
}

export function removepath(path: string): void {
  fs.unlinkSync(joinpath(path));
}
