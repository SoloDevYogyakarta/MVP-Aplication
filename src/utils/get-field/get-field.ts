import { GetFieldType } from 'src/types/get-field.type';
import { readpath } from '../system/system';

export function getField(filename?: GetFieldType) {
  try {
    if (filename === 'fake') {
      return readpath(`../../database/dataTxt/${filename}.txt`);
    }
    return JSON.parse(readpath(`../../database/dataTxt/${filename}.txt`));
  } catch (err) {
    throw new Error(err);
  }
}
