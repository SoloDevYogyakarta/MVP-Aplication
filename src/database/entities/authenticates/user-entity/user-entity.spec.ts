import { ModelCtor } from 'sequelize';
import { faker } from '@faker-js/faker';
import { userEntity, UserInstance } from './user-entity';
import { createpath } from '../../../../utils/system/system';
import { getField } from '../../../../utils/get-field/get-field';

describe('userEntity', () => {
  let entity: ModelCtor<UserInstance>;

  beforeEach(async () => {
    entity = await userEntity;
  });

  it('should to be defined', () => expect(entity).toBeDefined());

  it('render correctly', () => expect(entity).toMatchSnapshot());

  it('created new user', async () => {
    const username = faker.internet.userName();
    const created = await entity.create({
      username,
      password: 'password',
    });
    created.save();
    createpath('../../database/dataTxt/user-entity.txt', created);
    expect(created.username).toEqual(username);
  });

  try {
    it('updated', async () => {
      const { public_id } = getField('user-entity') as UserInstance;
      const findOne = await entity.findOne({ where: { public_id } });
      const email = faker.internet.email();
      findOne.username = `updated-${faker.internet.userName()}`;
      findOne.email = email;
      findOne.save();
      expect(findOne.email).toEqual(email);
    });

    it('findOne', async () => {
      const { public_id } = getField('user-entity') as UserInstance;
      const findOne = await entity.findOne({ where: { public_id } });
      expect(findOne.public_id).toEqual(public_id);
    });

    it('findAll', async () => {
      const findAll = await entity.findAll();
      expect(findAll.length).not.toEqual(0);
    });
  } catch (err) {
    // empty
  }
});
