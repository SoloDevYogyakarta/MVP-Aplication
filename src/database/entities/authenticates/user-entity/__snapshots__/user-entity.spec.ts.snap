// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userEntity render correctly 1`] = `[Function]`;
